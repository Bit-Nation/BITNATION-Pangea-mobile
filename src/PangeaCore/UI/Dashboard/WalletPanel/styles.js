// @flow

import { MediaQueryStyleSheet } from 'react-native-responsive';

import GlobalStyles from '../../../global/Styles';
import Colors from '../../../global/colors';

const styles = MediaQueryStyleSheet.create({
  ...GlobalStyles,
  panelViewTitle: {
    ...GlobalStyles.panelViewTitle,
    color: Colors.BitnationHighlightColor,
  },
  walletGridPanel: {
    ...GlobalStyles.gridPanelView,
    flex: 0,
    justifyContent: 'flex-start',
  },
  textWallet: { fontWeight: 'normal', color: '#FF8B00', fontSize: 30 },
});
export default styles;
