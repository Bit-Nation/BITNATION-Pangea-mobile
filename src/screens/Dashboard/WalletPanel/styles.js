import { MediaQueryStyleSheet } from 'react-native-responsive'
import { Dimensions } from 'react-native'

import GlobalStyles from '../../../global/Styles'

const styles = MediaQueryStyleSheet.create(
	{
		...GlobalStyles,
		// What is this?
		//list: { height: 44*5 },
	})
export default styles