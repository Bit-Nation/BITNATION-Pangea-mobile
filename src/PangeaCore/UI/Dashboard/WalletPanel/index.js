// @flow

import React from 'react';
import { View, Text } from 'react-native';
import _ from 'lodash';

import styles from './styles';

import PanelView from '../../../components/common/PanelView';
import i18n from '../../../global/i18n';
import { prettyWalletBalance } from '../../../utils/formatters';
import { resolveWallet } from '../../../utils/wallet';
import type { WalletType } from '../../../types/Wallet';

type Props = {
  /**
   * @desc Array of wallet objects.
   */
  +wallets: Array<WalletType>,
  /**
   * @desc Style to be applied to root view.
   */
  style: any,
};

/**
 * @desc Component to render wallet panel on dashboard
 * @return {React.Component} A component.
 */
const WalletPanel = ({ style, wallets } : Props) => {
  const patWallet = resolveWallet(wallets, 'XPAT');
  if (patWallet === null) {
    return <View />;
  }
  return (
    <View style={style}>
      <PanelView
        style={styles.walletGridPanel}
        childrenContainerStyle={styles.noflex}
        title={i18n.t('screens.dashboard.walletPanel.title')}
        titleStyle={styles.panelViewTitle}
      >
        {
          _.isEmpty(wallets)
            ?
              <Text style={styles.body}>{i18n.t('screens.dashboard.walletPanel.empty')}</Text>
            :
              <View>
                <Text style={styles.footnote}>{i18n.t('common.bitnationPat')}</Text>
                <Text style={[styles.largeTitle, styles.textWallet]}>{prettyWalletBalance(patWallet)}</Text>
              </View>
        }
      </PanelView>
    </View>
  );
};

WalletPanel.defaultProps = {
  wallets: [],
};

export default WalletPanel;
