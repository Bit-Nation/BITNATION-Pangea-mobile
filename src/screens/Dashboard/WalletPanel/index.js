// @flow

import React from 'react';
import { View, Text } from 'react-native';
import _ from 'lodash';

import styles from './styles';

import PanelView from '../../../components/common/PanelView';
import i18n from '../../../global/i18n';
import { prettyWalletBalance } from '../../../utils/formatters';
import type { WalletType } from '../../../types/Wallet';

type Props = {
  /**
   * @desc Array of wallet objects.
   */
  +wallets: Array<WalletType>,
  /**
   * @desc Style to be applied to root view.
   */
  style: any,
};

/**
 * @desc Component to render wallet panel on dashboard
 * @return {React.Component} A component.
 */
const WalletPanel = ({ style, wallets } : Props) => ((
  <View style={style}>
    <PanelView
      style={styles.walletGridPanel}
      childrenContainerStyle={styles.noflex}
      title={i18n.t('screens.dashboard.walletPanel.title')}
      titleStyle={styles.panelViewTitle}
    >
      {
        _.isEmpty(wallets)
          ?
            <Text style={styles.body}>{i18n.t('screens.dashboard.walletPanel.empty')}</Text>
          :
            <View>
              <Text style={styles.footnote}>XPAT</Text>
              <Text style={styles.currencyLarge}>{prettyWalletBalance(wallets[1])}</Text>
            </View>
      }
    </PanelView>
  </View>
));

WalletPanel.defaultProps = {
  wallets: [],
};

export default WalletPanel;
