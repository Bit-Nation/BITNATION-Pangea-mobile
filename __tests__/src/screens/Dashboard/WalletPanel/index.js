import React from 'react';
import { shallow } from 'enzyme';

import WalletPanel from '../../../../../src/screens/Dashboard/WalletPanel';

test('WalletPanel renders correctly', () => {
  const wrapper = shallow((
    <WalletPanel
      wallets={[]}
    />
  ));
  expect(wrapper).toMatchSnapshot();
});
