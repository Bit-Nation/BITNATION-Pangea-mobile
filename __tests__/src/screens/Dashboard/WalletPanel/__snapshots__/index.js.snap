// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPanel renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletPanel
    wallets={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
  },
}
`;
